### ISSUE TITLE
Path parameters getting trimmed when using conflicting route

### ISSUE THREAD

**Issue by NuroDev** [honojs/hono#1985]
### What version of Hono are you using?

3.12.4

### What runtime/platform is your app running on?

Cloudflare Workers

### What steps can reproduce the bug?

The following code example can help demonstrate the issue.

For context, we have a route that mimics the npm login endpoint (`/`) & up until now the following is the best implementation I have found to work (I'm terrible at regex).

```ts
import { type Context, Hono } from "hono";

const logParamsHandler = (c: Context) => c.json(c.req.param());

const PREFIX = "org.couchdb.user:";

const app = new Hono()
  .get(`/-/user/:${PREFIX}name`, logParamsHandler)
  .get("/:a/:b/:c", logParamsHandler)
  .get("/:a/:b/:c/:d", logParamsHandler);
```

### What is the expected behavior?

It seems since #1983 has been merged & subsequently version `3.12.4` has been published a complex route we use now no longer works.

If you make a request to the endpoint with the 3 path parameters (`/:a/:b/:c`) it will omit the first path parameter (`a`) / only return the final 2 parameters.

```bash
# /1/2/3
{
  "b": "2",
  "c": "3"
}
```

Similarly if you make a request to the endpoint with 4 path parameters (`/:a/:b/:c/:d`) it will show that it will only return the parameters for `c` & `d`.

```bash
# /1/2/3/4
{
  "c": "3",
  "d": "4"
}
```

### What do you see instead?

For the 2 example requests above I would expect to see the following, respectively

```bash
{
  "a": "1",
  "b": "2",
  "c": "3"
}
```

```bash
{
  "a": "1",
  "b": "2",
  "c": "3",
  "d": "4"
}
```

### Additional information

This seems like a minor regression that is causing the "login" route (`/-/user/:${PREFIX}name`) to pin the max length or number of parameters that can be returned & resulting it it (somehow) trimming it to the final 2 parameters.

**Comment by yusukebe** [comment]
Hi @NuroDev !

This is a bug. I'll investigate it. Thanks.

**Comment by yusukebe** [comment]
@NuroDev 

I've fixed this issue and released a new version, v3.12.5, which includes the change. Please try it.

**Comment by NuroDev** [comment]
Yup that fixed the issue @yusukebe. Thank you so much!
